eth/catalyst/api.go

```diff
@@ -95,14 +103,20 @@ type ConsensusAPI struct {
 
 // NewConsensusAPI creates a new consensus api for the given backend.
 // The underlying blockchain needs to have a valid terminal total difficulty set.
-func NewConsensusAPI(eth *eth.Ethereum) *ConsensusAPI {
+func NewConsensusAPI(eth *eth.Ethereum, newpayloadTimeout time.Duration) *ConsensusAPI {
 	if eth.BlockChain().Config().TerminalTotalDifficulty == nil {
 		log.Warn("Engine API started but chain not configured for merge yet")
 	}
+	// Sanitize the timeout config for creating payload.
+	if newpayloadTimeout < time.Millisecond*100 {
+		log.Warn("Sanitizing new payload timeout to default", "provided", newpayloadTimeout, "updated", ethconfig.Defaults.Miner.NewPayloadTimeout)
+		newpayloadTimeout = ethconfig.Defaults.Miner.NewPayloadTimeout
+	}
```

Comment:
Isn't it a bit odd if `101ms` -> `101ms`, but `99ms` -> `2s` ? 
How about something like this?
```suggestion
	if newpayloadTimeout == 0 {
		log.Warn("Sanitizing new payload timeout to default", "provided", newpayloadTimeout, "updated", ethconfig.Defaults.Miner.NewPayloadTimeout)
		newpayloadTimeout = ethconfig.Defaults.Miner.NewPayloadTimeout
	}
	if newpayloadTimeout < time.Millisecond*100 {
		log.Warn("Low payload timeout may cause high amount of empty blocks", "provided", newpayloadTimeout, "default", ethconfig.Defaults.Miner.NewPayloadTimeout)
	}
```
If someone does decide to set very small timeout causing empty blocks, well, it's their node...

Reply:
yup, good idea